src/book/views.py

```diff
@@ -33,7 +33,18 @@ def form_valid(self, form):
     def get_success_url(self):
         return reverse_lazy('book:book-detail', kwargs={'pk': self.object.book.pk})
 
-
+class ShowComments(BaseTemplatePageMixin, generic.ListView):
+    model = models.BookComments
+    template_name = 'book/comments_list.html'
+
+class SearchCom(BaseTemplatePageMixin, generic.TemplateView):
```

Comment:
You do not need a separate view and template for to show the search results.

First improvement you can do is to use the same template for showing all comments and showing the comments searched comments. So here you can just use `book/comments_list.html` template.

Another improvement that you do not need this view at all. You can use `ShowComments` view which will accept the request parameter `search` and override the `get_queryset` method with correct filter by `search param`. Example of overriding is present here https://medium.com/@hassanraza/when-to-use-get-get-queryset-get-context-data-in-django-952df6be036a

Reply:
fixed